### PR TITLE
doc: add a README to explain the project

### PR DESCRIPTION
**Pull request by ccoVeille**

### CODE REVIEW

**Comment by ccoVeille** [comment]
Hi @ldemailly

may I ask you to review

File: README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Go SafeCast
+
+## Origin of this project
+
+In Go, integer type conversion can lead to unexpected behavior and errors if not handled carefully.
+When converting between integer types, issues can happen when converting between signed and unsigned integers, or when converting to a smaller integer type. The code will panic.
```
Comment (ldemailly) [inline README.md:6]: When does it ever panic? can you add a play.go.dev link of code that does and/or an example?

It seems a bit repetitive "unexpected behavior and errors" isn't it the same?

Comment (ccoVeille) [reply README.md]: You are right, it doesn't panic. I updated the README

File: README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+func main() {
+  var a uint64
+  a = 2147483647
+  b := int32(a)
+  fmt.Println(b) // 2147483647
+
+  a = 2147483647 + 1
+  b = int32(a)
+  fmt.Println(b) // -2147483648 Stack overflow
```
Comment (ldemailly) [inline README.md:51]: it's not a stack overflow

Comment (ccoVeille) [reply README.md:51]: fixed with https://github.com/ccoVeille/go-safecast/pull/8 because I forgot to commit locally before pushing :facepalm:

**Pull request merged**